Hw13/main.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <numeric>
+#include <algorithm>
+#include <span> 
+
+constexpr size_t MarksCount{ 4 };
+
+struct Student
+{
+    const char* name;
+    int marks[MarksCount];
+};
+
+std::ostream &operator<<(std::ostream &os, const Student &st)
+{
+    os << "Age: " << st.name << "\n";
+
+    os << "Marks: ";
+    for (const auto& n : st.marks)
+        os << n << " ";
+
+    os << "\n";
+    return os;
+}
+
+void task1();
+void task2();
+void task3();
+
+int main() 
+{
+    task1();
+    task2();
+    task3();
+}
+
+double calcStudentAverage(const Student &student) 
+{
+    return std::reduce(student.marks, student.marks + MarksCount, 0.0) / MarksCount;
+}
+
+void task1() 
+{
+    constexpr Student st{ "Leon",{1,2,3,4}};
+    std::cout << "Average mark is: " << calcStudentAverage(st) << "\n";
+}
+
+bool sortStudents(const Student &left, const Student &right)
```

Comment:
the name is somewhat ambigious) simple `compare` or `greaterThan` seems more natural)

Reply:
agree, my bad